src/preferences/settingsmanager.cpp

```diff
@@ -14,6 +55,10 @@ SettingsManager::SettingsManager(const QString& settingsPath)
         QDir().mkpath(settingsPath);
     }
 
+    if (!checkSettingsPathPerm(settingsPath)) {
+        exit(EXIT_FAILURE);
```

Comment:
I'm wondering if it wouldn't be best to `qApp->quit()` or `qApp->exit(EXIT_FAILURE)`. Is there no risk to have a raw exit to leave certain resource in stale state such as the DB or any devices? Wdyt?